ENH: GH11128 add weekday_name to DatetimeIndex and .dt

BastiaanBergman opened the pull request:
closes #11128 
replaces #11813
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Completed merge of Bahrunnur's PR 'weekday-name'
    Moved 'enhancements' to v0.18.1.txt
    Added tests to pandas/tests/series/test_datetime_values.py
    Added weekday_name property producing 'NaT' to NaTType class

bahrunnur committed:
ENH: GH11128 add weekday_name to DatetimeIndex and .dt

	Completed merge of Bahrunnur's PR 'weekday-name'
	Moved 'enhancements' to v0.18.1.txt
	Added tests to pandas/tests/series/test_datetime_values.py
	Added weekday_name property producing 'NaT' to NaTType class
 Author:    BastiaanBergman <Bastiaan.Bergman@gmailcom>

jreback commented on doc/source/api.rst:
```diff
@@ -469,6 +469,7 @@ These can be accessed like ``Series.dt.<property>``.
    Series.dt.days_in_month
    Series.dt.tz
    Series.dt.freq
+   Series.dt.weekday_name
```
can you move after ``.weekday``

jreback commented on doc/source/api.rst:
```diff
@@ -1487,6 +1488,7 @@ Time/Date Components
    DatetimeIndex.is_year_start
    DatetimeIndex.is_year_end
    DatetimeIndex.inferred_freq
+   DatetimeIndex.weekday_name
```
same here

jreback commented on pandas/tseries/tests/test_timeseries.py:
```diff
@@ -3541,6 +3541,15 @@ def test_datetimeindex_accessors(self):
 
+        # GH 11128
+        self.assertEqual(dti.weekday_name[4], u'Monday')
+        self.assertEqual(dti.weekday_name[5], u'Tuesday')
```
can you add a test for ``Timestamp`` (see where tests for ``.weekday`` are, prob in ``tseries/test_timeseries.py``

jreback commented on pandas/tslib.pyx:
```diff
@@ -4390,6 +4395,38 @@ def get_start_end_field(ndarray[int64_t] dtindex, object field, object freqstr=N
+
+    _dayname = np.array(
+        ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
+         dtype=np.str )
```
``dtype=np.object_`` is more consistent with what we use.

jreback commented on pandas/tseries/tests/test_base.py:
```diff
@@ -32,7 +32,7 @@ def test_ops_properties(self):
                                    'is_month_start', 'is_month_end',
                                    'is_quarter_start',
                                    'is_quarter_end', 'is_year_start',
-                                   'is_year_end'],
```
there are addl tests in ``tests/series/test_datetime_values.py`` (just add the new propery name)

BastiaanBergman replied:
Yes, and they were added to in my original PR.

jreback commented:
looks good. just some minor points.

come to think of it, prob should add ``month_name`` as well I guess. Then of course there is the locale issue. Can you make an issue for these 2 things?

jreback commented:
I *think* we can just instantiate the ``LocaleTime()`` object when this module is imported. Not really sure why that wouldn't work, you can try it if you want. This should really be a singleton type object, e.g.

```
locale_time = LocalTime()
```

then use ``locale_time``

Commit:
Small fixes as requested by jreback, GH12803.

-Moved to group .weekday_name next to .weekday in docs
-Added Timestamp test to tseries/tests/test_timeseries.py
-Change dtype of weekday_name to np.object_

jreback commented on pandas/tests/series/test_datetime_values.py:
```diff
@@ -25,7 +25,7 @@ class TestSeriesDatetimeValues(TestData, tm.TestCase):
     def test_dt_namespace_accessor(self):
 
-        # GH 7207
+        # GH 7207, 11128
```
I mean add it here as well

jreback commented on pandas/tslib.pyx:
```diff
@@ -4411,7 +4411,7 @@ def get_date_name_field(ndarray[int64_t] dtindex, object field):
     _dayname = np.array(
         ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
-         dtype=np.str )
+         dtype=np.object_ ) # was np.str
```
remove the comment

Commit:
Tiny fixes as requested by jreback, GH12803.

jreback commented:
ok this looks good. I'd like to use the ``LocaleTime()`` object as I described (should be cached) to return the locale specific names here.

jreback commented:
xref #12806

jreback commented:
can you try adding this with the ``LocaleTime``support?

BastiaanBergman commented:
Yes, just need to find the time, maybe later this week.
On Apr 17, 2016 8:46 AM, "Jeff Reback" <notifications@github.com> wrote:

> can you try adding this with the LocaleTimesupport?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/12803#issuecomment-211045629>
>

jreback commented:
@BastiaanBergman ok, great thanks

jreback commented:
@jorisvandenbossche comments?

I guess we can address using ``LocaleTime()`` later.

BastiaanBergman commented:
Sorry, didn't get to it, will try again late May.

jorisvandenbossche commented:
Looks good

jreback commented:
thanks!
